src/main/scala/co/datamechanics/delight/DelightStreamingConnector.scala

```diff
@@ -127,7 +129,7 @@ class DelightStreamingConnector(sparkConf: SparkConf) extends Logging {
     * - Uses a dedicated httpClient to avoid collision with the one used by the main thread
     * - Payload is a Json Object containing the dm_app_id
     */
-  def sendHeartbeat(): Unit = {
+  def sendHeartbeat(): Unit = Utils.time(shouldLogDuration, "sendHeartbeat") {
```

Comment:
The more I look at the change the often I ask myself this question: why not to use this `shouldLogDuration` inside `Utils.time` since that's the only reason for this `shouldLogDuration`. It's irrelevant to the class.

Reply:
You're right it's irrelevant to the class!
The reason why I did this is that `shouldLogDuration` is determined by the `sparkConf`, which is not accessible in the `Utils` package.

So an alternative would be to pass the `sparkConf` to `Utils.time` instead of passing it `shouldLogDuration`, but I'm not sure it's better?

Reply:
I've got the same doubt. And I've got no other option, but as a code reader I think this change would make it slightly better.

Reply:
I'll postpone this one. If every call to `time` needs to extract the value of `shouldLogDuration` from the `sparkConf`, this will add some overhead to the calls to `time`.
A solution could be to create a `Timer` object that extracts the value from `sparkConf` at initialization time.